tests/gateway/test-nodeport.yaml

```diff
@@ -0,0 +1,61 @@
+replicaCount: 1
+
+image:
+  # -- Repository location of the Trino Gateway image, typically `organization/imagename`
+  repository: "trinodb/trino-gateway"
+  pullPolicy: IfNotPresent
+
+command:
+  - "/bin/sh"
+  - "-c"
+  - |
+    cat /etc/certificates/tls.crt /etc/certificates/tls.key > /etc/scratch/tls.pem && \
+    java -XX:MinRAMPercentage=80.0 -XX:MaxRAMPercentage=80.0 -jar /usr/lib/trino/gateway-ha-jar-with-dependencies.jar /etc/gateway/config.yaml
+
+config:
+  serverConfig:
+    node.environment: test
+    http-server.http.enabled: true
+    http-server.https.enabled: true
+    http-server.https.port: 8443
+    http-server.https.keystore.path: /etc/scratch/tls.pem
+  dataStore:
+    # The connection details for the backend database for Trino Gateway and Trino query history
+    jdbcUrl: jdbc:postgresql://gateway-backend-db-postgresql.postgres-gateway.svc.cluster.local:5432/gateway
+    user: gateway
+    password: pass0000
+
+service:
+  spec:
+    type: NodePort
+    ports:
+      - protocol: TCP
+        name: request
+        nodePort: 30443
+      - protocol: TCP
+        name: gateway-http
+        nodePort: 30080
+        port: 8080
+        targetPort: 8080
+
+volumes:
+  - name: certificates
+    secret:
+      secretName: certificates
+  - name: scratch
+    emptyDir:
+      sizeLimit: 10Mi
+
+volumeMounts:
+  - name: certificates
+    mountPath: /etc/certificates
+  - name: scratch
+    mountPath: /etc/scratch
+
+resources:
```

Comment:
Remove properties not required for the test, especially if they're already tested in other suites.